WcaOnRails/app/webpacker/javascript/competition_results/index.js

```diff
@@ -0,0 +1,112 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import { Table, Popup } from 'semantic-ui-react';
+import useLoadedData from '../requests/loadable';
+import { registerComponent } from '../wca/react-utils';
+import Loading from '../requests/Loading';
+import Errored from '../requests/Errored';
+import { formatAttemptResult } from '../wca-live/attempts';
+
+import EventIcon from '../wca/EventIcon';
+import CountryFlag from '../wca/CountryFlag';
+import events from '../wca/events.js.erb';
+import './index.scss';
+
+const CompetitionResultsNavigation = ({ eventIds, selected, onSelect }) => (
+  <div className="events-list">
+    {eventIds.map((event, index) => (
+      <Popup
+        key={event}
+        content={events.byId[event].name}
+        trigger={(
+          <EventIcon
+            key={event}
+            id={event}
+            onClick={() => onSelect(index)}
+            className={classnames(selected === index && 'selected')}
+          />
+          )}
+        inverted
+      />
+    ))}
+  </div>
+);
+
+const RoundResultsTable = ({ round, eventName, eventId }) => (
+  <>
+    <h2>{`${eventName} ${round.name}`}</h2>
+    <Table>
+      <Table.Header>
+        <Table.Row>
+          <Table.HeaderCell>#</Table.HeaderCell>
+          <Table.HeaderCell>Name</Table.HeaderCell>
+          <Table.HeaderCell>Best</Table.HeaderCell>
+          <Table.HeaderCell>Average</Table.HeaderCell>
+          <Table.HeaderCell>Citizen Of</Table.HeaderCell>
+          <Table.HeaderCell>Solves</Table.HeaderCell>
+        </Table.Row>
+      </Table.Header>
+      <Table.Body>
+        {round.results.map((result) => (
+          <Table.Row key={result.id}>
+            <Table.Cell>{result.pos}</Table.Cell>
+            <Table.Cell>
+              <a href={`/persons/${result.wca_id}`}>{`${result.name}`}</a>
```

Comment:
I've started a "routes" module [here](https://github.com/thewca/worldcubeassociation.org/pull/5340/files#diff-40e8a9d588d2ad70187890336dd4bb09), would you mind adding the various website urls you used in this component there?

The rationale behind this is:
  - to group the routes used in js in a common place
  - but most importantly to make sure to rely on rails to provide us with the appropriate path, in case some day it's slightly altered (in the specific case of `persons` I highly doubt it will ever happens, but you get the point).

To give you an example, a helper for that route could look like:
```js
export const personUrl = (wcaId) => {
  return `<%= URI.decode(Rails.application.routes.url_helpers.person_path("${wcaId}")) %>`;
};
```
(please note I did *not* run the code, apologies if it's slightly off!)

Reply:
@viroulep  Would you like me to add the API paths like this as well? If so, could you guide me on where I can access them through Rails? I'm unable to figure that out.

Reply:
Yes and yes, will try to give you an example when I can!

Reply:
If you want to get a link for this for instance:
```
  const { loading, error, data } = useLoadedData(
    `/api/v0/competitions/${competitionId}/results/${eventId}`
  );
```
You would probably want to expose a route like that:
```js
export const competitionResultsApiUrl = (id, eventId) => `<%= URI.decode(Rails.application.routes.url_helpers.api_v0_competition_results_path("${id}", "${eventId}")) %>`;
```

I tested this locally and it didn't work because somehow adding `(/:eventId)` to the route prevent rails from generating its name automatically :s
For this to work you will have to change this route:
```ruby
        get '/results(/:event_id)' => 'competitions#results'
```
to
```ruby
        get '/results(/:event_id)' => 'competitions#results', as: :results
```

You can get a list of all routes and their generated name by running `bin/rails routes`; alternatively you can also start a rails console with `bin/rails c` and take a look at `Rails.application.routes.url_helpers.methods` which includes all routes exposed by rails.